### PR TITLE
added MovieListPage component, improved Search, Filter, Search, improved styles

### PR DESCRIPTION
**Pull request by alex-lis-dev**

### CODE REVIEW

File: src/components/MovieDetails/MovieDetails.jsx [hunk 1]
```diff
@@ -2,9 +2,12 @@ import React from "react";
 import "./MovieDetails.css";
 import PropTypes from "prop-types";
 
-const MovieDetails = ({ movie = null }) => {
+const MovieDetails = ({ movie = null, handleSearchIconClick = () => {} }) => {
```
Comment (hicugi) [inline src/components/MovieDetails/MovieDetails.jsx]: You can't have `movie = null` and render a property from it.
Example: `movie.title`, `movie.poster_path` etc.

Rename the `handleSearchIconClick` to `onSearchClick` and don't assign default function into it.

File: src/components/MovieForm/MovieForm.jsx [hunk 2]
```diff
@@ -17,91 +19,121 @@ import {
...
+      [key]: value,
+    });
+  };
+
   const handleSubmit = (event) => {
     event.preventDefault();
-    const formData = new FormData(event.target);
-    const movieData = Object.fromEntries(formData);
-    onSubmit(movieData);
+    //const formData = new FormData(event.target);
```
Comment (hicugi) [inline src/components/MovieForm/MovieForm.jsx]: Remove unnecessary comments

File: src/components/MovieForm/components/AddAndEditMovieDialog/AddAndEditMovieDialog.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import Dialog from "../../../Dialog/Dialog";
+import MovieForm from "../../MovieForm";
+import { AddMovieText } from "../../../../helpers/constants";
+import PropTypes from "prop-types";
+
+const AddAndEditMovieDialog = ({ isOpen, onClose, movieToEdit }) => {
+  const handleFormSubmit = (movieData) => {
+    console.log(movieData);
```
Comment (hicugi) [inline src/components/MovieForm/components/AddAndEditMovieDialog/AddAndEditMovieDialog.jsx]: remove

File: src/components/MovieForm/components/AddAndEditMovieDialog/AddAndEditMovieDialog.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    onClose();
+  };
+
+  return (
+    isOpen && (
+      <Dialog title={AddMovieText} onClose={onClose}>
```
Comment (hicugi) [inline src/components/MovieForm/components/AddAndEditMovieDialog/AddAndEditMovieDialog.jsx]: You can pass the `isOpen` to `Dialog` component with prop like `active`

File: src/components/MovieForm/components/DeleteMovieDialog/DeleteMovieDialog.jsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+} from "../../../../helpers/constants";
+
+const DeleteMovieDialog = ({ isOpen, onClose, movieToDelete }) => {
+  const handleDeleteMovie = () => {
+    console.log(movieToDelete.title);
+    onClose();
+  };
+
+  return (
+    isOpen && (
```
Comment (hicugi) [inline src/components/MovieForm/components/DeleteMovieDialog/DeleteMovieDialog.jsx]: So you don't have to repeat the same behaviour here

File: src/components/MovieListPage/MovieListPage.jsx [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+import AddAndEditMovieDialog from "../MovieForm/components/AddAndEditMovieDialog/AddAndEditMovieDialog.jsx";
+
+const MovieListPage = () => {
+  const genres = [
+    "All",
+    "Documentary",
+    "Horror",
+    "War",
+    "MUSIC",
+    // ...new Set(
```
Comment (hicugi) [inline src/components/MovieListPage/MovieListPage.jsx]: remove

File: src/components/MovieListPage/MovieListPage.jsx [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    //   orderedMovies.reduce((acc, movie) => acc.concat(movie.genres), [])
+    // ),
+  ];
+
+  const [orderedMovies, setOrderedMovies] = useState([]);
+  const [selectedMovie, setSelectedMovie] = useState();
```
Comment (hicugi) [inline src/components/MovieListPage/MovieListPage.jsx]: The `selectedMovie` should be `null` by default

**Review by hicugi — CHANGES_REQUESTED** [review]
~~Major issue - abortion of the search request is not working~~

The default value for `movie` should be `null`

File: src/components/MovieDetails/MovieDetails.jsx (hunk 1 above)
Comment (alex-lis-dev) [reply src/components/MovieDetails/MovieDetails.jsx]: removed default values, corrected naming

File: src/components/MovieForm/components/AddAndEditMovieDialog/AddAndEditMovieDialog.jsx (hunk 3 above)
Comment (alex-lis-dev) [reply src/components/MovieForm/components/AddAndEditMovieDialog/AddAndEditMovieDialog.jsx]: removed

File: src/components/MovieForm/MovieForm.jsx (hunk 2 above)
Comment (alex-lis-dev) [reply src/components/MovieForm/MovieForm.jsx]: removed

File: src/components/MovieForm/components/AddAndEditMovieDialog/AddAndEditMovieDialog.jsx (hunk 4 above)
Comment (alex-lis-dev) [reply src/components/MovieForm/components/AddAndEditMovieDialog/AddAndEditMovieDialog.jsx]: corrected

File: src/components/MovieForm/components/DeleteMovieDialog/DeleteMovieDialog.jsx (hunk 5 above)
Comment (alex-lis-dev) [reply src/components/MovieForm/components/DeleteMovieDialog/DeleteMovieDialog.jsx]: corrected

File: src/components/MovieListPage/MovieListPage.jsx (hunk 7 above)
Comment (alex-lis-dev) [reply src/components/MovieListPage/MovieListPage.jsx]: done

File: src/components/MovieListPage/MovieListPage.jsx (hunk 6 above)
Comment (alex-lis-dev) [reply src/components/MovieListPage/MovieListPage.jsx]: removed

**Comment by alex-lis-dev** [comment]
The corrections are applied, console logs and comments are removed

**Pull request merged**